### ISSUE TITLE
Code in wrong place

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1356]
**Reported by anonymous on 16 May 2014 17:40 UTC**

Hello,

I have notice a part of code in wrong place on airodump-ng...

airodump-ng.c > int dump_add_packet function:

```
/* find wpa handshake */
    if( h80211[0] == 0x10 )
    {
        /* reset the WPA handshake state */

        if( st_cur != NULL && st_cur->wpa.state != 0xFF )
            st_cur->wpa.state = 0;
//        printf("initial auth %d\n", ap_cur->wpa_state);
    }
```

You check "st_cur != NULL" ... but "st_cur" is always "NULL" at this place !

Enjoy.

**Comment by aircrack-ng** [comment]
**Modified by misterx on 16 May 2014 17:40 UTC**

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 16 May 2014 17:40 UTC**

Oups i'm very sorry, it's not "airmon-ng" but "airodump-ng" (airodump-ng.c) file !

**Comment by aircrack-ng** [comment]
**Comment by misterx on 16 May 2014 17:40 UTC**

Which version are you using? 1.1? Svn trunk? 1.2 beta 3?

**Comment by aircrack-ng** [comment]
**Modified by misterx on 16 May 2014 17:40 UTC**

**Comment by aircrack-ng** [comment]
**Comment by moshe on 16 May 2014 17:40 UTC**

Confirmed and fixed in aircrack-ng/aircrack-ng@a0cd0db3215f0bc66dade0748591ed625dc86394
